### PR TITLE
Inbox notification: add client unit tests

### PR DESCRIPTION
**Pull request by octaedro**
Fixes #partially-4099

This PR adds:
- Unit tests on client-side.

> **_NOTE:_**  As this is a part of a big functionality this branch will be merged with the branch `add/4099`, and it will be that branch that will be merged with `master`.
To make the code reviewing easier, this PR only has the code referring to the functionality described in the title.
If you want to test everything together, all the changes are merged in the branch: `add/4099_test_branch`.

### Detailed test instructions:

- Run unit tests: `npm run test`
- Verify there aren't errors.

### Changelog Note:
Enhancement: Added client unit tests to inbox notifications.

### CODE REVIEW

File: tests/api/admin-notes.php [hunk 1]
```diff
@@ -314,5 +316,109 @@ public function test_get_notes_schema() {
...
+	 */
+	public function test_delete_single_note_without_permission() {
+		$response = $this->server->dispatch( new WP_REST_Request( 'DELETE', $this->endpoint . '/delete/3' ) );
+		$note     = $response->get_data();
+
+		$this->assertEquals( 401, $response->get_status() );
+	}
+
+	/**
+	 * Test undiong a single note delete.
```
Comment (becdetat) [inline tests/api/admin-notes.php:354]: Typo

File: tests/api/admin-notes.php [hunk 2]
```diff
@@ -314,5 +316,109 @@ public function test_get_notes_schema() {
...
+				'is_deleted' => '0',
+			)
+		);
+
+		$response = $this->server->dispatch( $request );
+		$note     = $response->get_data();
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( false, $note['is_deleted'] );
+	}
+
```
Comment (becdetat) [inline tests/api/admin-notes.php:375]: These tests are great :-D

**Comment by becdetat** [comment]
I couldn't get the PHP unit tests to run without using the update in 860127e - WooCommerce changed the base path of the unit test base class a few weeks ago. You might just be able to cherry-pick that commit. Everything tested ok though.

File: tests/api/admin-notes.php (hunk 1 above)
Comment (octaedro) [reply tests/api/admin-notes.php]: Cool, this was changed in the commit [be254b0](https://github.com/woocommerce/woocommerce-admin/pull/4386/commits/be254b001ef0acbb2724ef93a337f56eaeb962b3)

File: client/header/activity-panel/panels/inbox/test/index.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
```
Comment (jeffstieler) [inline client/header/activity-panel/panels/inbox/test/index.js:4]: Didn't we decide to use React Testing Library instead?

Comment (octaedro) [reply client/header/activity-panel/panels/inbox/test/index.js]: You're right @jeffstieler. I removed `Enzyme` and added `React Testing Library` in the commit 68be4dc

File: client/header/activity-panel/panels/inbox/test/index.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+	test( 'should render a notification type thumbnail', () => {
+		note.layout = 'thumbnail';
+		const { container } = render(
+			<InboxNoteCard
+				key={ note.id }
+				note={ note }
+				lastRead={ lastRead }
+			/>,
+			{
+				disableLifecycleMethods: true,
```
Comment (jeffstieler) [inline client/header/activity-panel/panels/inbox/test/index.js:65]: Curious - why was this needed for this test?

Comment (octaedro) [reply client/header/activity-panel/panels/inbox/test/index.js]: Thanks Jeff. Yes, it's unnecessary. It was removed in the commit 8d6d9a0
